drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -3804,7 +3822,82 @@ static void rswitch_fwd_init(struct rswitch_private *priv)
 
 	/* Enable SC-Tag filtering mode for VLANs */
 	rs_write32(BIT(1), priv->addr + FWGC);
+
+#if IS_ENABLED(CONFIG_RENESAS_FWD_MON)
+	// CPU mirroring
+	rs_write32(priv->mon_chain | (RSWITCH_HW_NUM_TO_GWCA_IDX(priv->gwca.index) << 16),
+	   priv->addr + FWCMPTC);
+#endif
+}
+
+#if IS_ENABLED(CONFIG_RENESAS_FWD_MON)
+static int rswitch_mon_ndev_allocate(struct rswitch_private *priv)
+{
+	struct platform_device *pdev = priv->pdev;
+	struct net_device *ndev;
+	struct rswitch_device *rdev;
+	int err;
+
+	ndev = alloc_etherdev_mqs(sizeof(struct rswitch_device), 1, 1);
+	if (!ndev)
+		return -ENOMEM;
+
+	SET_NETDEV_DEV(ndev, &pdev->dev);
+	ether_setup(ndev);
+
+	rdev = netdev_priv(ndev);
+	rdev->ndev = ndev;
+	rdev->priv = priv;
+	INIT_LIST_HEAD(&rdev->routing_list);
+	mutex_lock(&priv->rdev_list_lock);
+	list_add_tail(&rdev->list, &priv->rdev_list);
+	mutex_unlock(&priv->rdev_list_lock);
+	rdev->port = -1;
+	rdev->etha = NULL;
+	rdev->remote_chain = 0;
+	rdev->addr = priv->addr;
+
+	spin_lock_init(&rdev->lock);
+
+	ndev->features = NETIF_F_RXCSUM;
+	ndev->hw_features = NETIF_F_RXCSUM;
+	ndev->base_addr = (unsigned long)rdev->addr;
+	snprintf(ndev->name, IFNAMSIZ, "rmon");
+	ndev->netdev_ops = &rswitch_netdev_ops;
+
+	netif_napi_add(ndev, &rdev->napi, rswitch_poll, 64);
+
+	eth_hw_addr_random(ndev);
```

Comment:
Do we really need an MAC address for the monitoring interface?

Moreover, it will be good to configure it in a such way, that Linux would not try to use to sent packets

Reply:
I tested with zero addresses, but unfortunately in this case I got an error while link up. Also, I didn't find the proper solution to not allow Linux to send packets to interface. So, I leave this line as in the previous commit. If you have some idea how to to this, please, let me know - I will check your suggestion.